pkg/cmd/task/logs_test.go

```diff
@@ -0,0 +1,519 @@
+				tb.PodSpec(
+					tb.PodContainer("step1", "step1:latest"),
+					tb.PodContainer("step2", "step2:latest"),
+				),
+				cb.PodStatus(
+					cb.PodPhase(corev1.PodSucceeded),
+				),
+			),
+		},
+	})
+
+	logs := fake.Logs(
+		fake.Task("pod",
+			fake.Step("step1", "step1 log"),
+			fake.Step("step2", "step2 log"),
+		),
+	)
+
+	testParams := []struct {
+		name      string
+		limit     int
+		last      bool
+		namespace string
+		input     pipelinetest.Clients
+		prompt    htest.PromptTest
+	}{
+		{
+			name:      "Get all input",
+			limit:     5,
+			last:      false,
+			namespace: "ns",
+			input:     cs,
+			prompt: htest.PromptTest{
+				CmdArgs: []string{},
+				Procedure: func(c *goexpect.Console) error {
+					if _, err := c.ExpectString("Select task:"); err != nil {
+						return err
+					}
+
+					if _, err := c.SendLine(string(terminal.KeyArrowDown)); err != nil {
+						return err
+					}
+
+					if _, err := c.ExpectString("task"); err != nil {
+						return err
+					}
+
+					if _, err := c.SendLine(string(terminal.KeyEnter)); err != nil {
+						return err
+					}
+
+					if _, err := c.ExpectString("Select taskrun:"); err != nil {
+						return err
+					}
+
+					if _, err := c.SendLine(string(terminal.KeyArrowDown)); err != nil {
+						return err
+					}
+
+					if _, err := c.ExpectString("taskrun2 started 3 minutes ago"); err != nil {
+						return err
+					}
+
+					if _, err := c.SendLine(string(terminal.KeyArrowDown)); err != nil {
+						return err
+					}
+
+					if _, err := c.ExpectString("taskrun1 started 5 minutes ago"); err != nil {
+						return err
+					}
+
+					if _, err := c.SendLine(string(terminal.KeyArrowUp)); err != nil {
+						return err
+					}
+
+					if _, err := c.ExpectString("taskrun2 started 3 minutes ago"); err != nil {
+						return err
+					}
+
+					if _, err := c.SendLine(string(terminal.KeyEnter)); err != nil {
+						return err
+					}
+
+					return nil
+				},
+			},
+		},
+		{
+			name:      "Specify task name and choice task run name from interactive menu",
+			limit:     5,
+			last:      false,
+			namespace: "ns",
+			input:     cs,
+			prompt: htest.PromptTest{
+				CmdArgs: []string{"task"},
+				Procedure: func(c *goexpect.Console) error {
+					if _, err := c.ExpectString("Select taskrun:"); err != nil {
+						return err
+					}
+
+					if _, err := c.SendLine(string(terminal.KeyArrowDown)); err != nil {
+						return err
+					}
+
+					if _, err := c.ExpectString("taskrun2 started 3 minutes ago"); err != nil {
+						return err
+					}
+
+					if _, err := c.SendLine(string(terminal.KeyArrowDown)); err != nil {
+						return err
+					}
+
+					if _, err := c.ExpectString("taskrun1 started 5 minutes ago"); err != nil {
+						return err
+					}
+
+					if _, err := c.SendLine(string(terminal.KeyEnter)); err != nil {
+						return err
+					}
+
+					return nil
+				},
+			},
+		},
+		{
+			name:      "Specify task name and limit as 2",
+			limit:     2,
```

Comment:
Would this test be better with `limit: 1` to assess whether limit returns less than the full amount?